### PR TITLE
Add a troubleshooting section for IoT

### PR DESCRIPTION
**Pull request by ctron**
### Type of change

- Documentation

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+iot.enmasse.io/deviceRegistryCleanup
+----
+If the list contains an entry of `iot.enmasse.io/deviceRegistryCleanup`, then resource cleanup process
+is still pending.
+
+. Manually remove the finalizer `iot.enmasse.io/deviceRegistryCleanup` from the list of finalizers:
++
+[options="nowrap",subs="attributes, +quotes"]
+----
+{cmdcli} edit iotproject iot -n myapp
```
Comment (dejanb) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]: Maybe we can document an easy copy/paste solutions in addition

```{cmdcli} patch iotproject iot -p '{"metadata":{"finalizers": []}}' --type=merge```

Comment (ctron) [reply documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]: Yea … the problem with that is that you are actually deleting all finalizers. Which is a difference to simply removing one. Not sure that makes a big difference in this case though.

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+The output should show the project in the state "Terminating".
+In addition, verify that the cleanup finalizer is still present:
++
+[options="nowrap",subs="attributes, +quotes"]
+----
+{cmdcli} get iotproject iot -n myapp -ojsonpath='{range .metadata.finalizers[*]}{..}{"\n"}{end}'
+iot.enmasse.io/deviceRegistryCleanup
+----
+If the list contains an entry of `iot.enmasse.io/deviceRegistryCleanup`, then resource cleanup process
+is still pending.
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]: Question: Does this mean the user needs to wait until it is no longer pending to proceed? Or is this the state it needs to be in to proceed to the next step? 
If I'm following the logic correctly, I think it's the latter, so I added a clause below to clarify.

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+. Manually remove the finalizer `iot.enmasse.io/deviceRegistryCleanup` from the list of finalizers:
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]:
```suggestion
. Locate the finalizer `iot.enmasse.io/deviceRegistryCleanup` in the list of finalizers:
```

File: documentation/assemblies/assembly-iot-service-admin-guide.adoc [hunk 4]
```diff
@@ -16,3 +16,7 @@ include::../modules/proc-iot-installing-services.adoc[leveloffset=+1]
 include::../modules/ref-iot-config-example.adoc[leveloffset=+2]
 
 include::../modules/ref-iot-config-example-adapters.adoc[leveloffset=+2]
+
+== Troubleshooting guide
```
Comment (jenmalloy) [inline documentation/assemblies/assembly-iot-service-admin-guide.adoc]:
```suggestion
== Troubleshooting 
```

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 5]
```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// assembly-service-admin.adoc
+
+[id='proc-iot-troubleshoot-stuck-iotproject-{context}']
+= Fix `IoTProject` stuck in termination
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]:
```suggestion
= Resolving an IoT project deletion stuck in terminating state
```

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 6]
```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// assembly-service-admin.adoc
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]:
```suggestion
// assembly-iot-service-admin-guide.adoc
```

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+In the case that the infrastructure is not expected to function ever again, it might be desirable
+to force the destruction of the `IoTProject` resources.
+
+WARNING: Manually removing the resource cleanup finalizer will skip the cleanup process,
+ and prevent the system from properly cleaning up.
+
+.Procedure
+
+. Evaluate if the project is stuck in the termination state using the `{cmdcli}` tool:
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]:
```suggestion
. Determine if the project is stuck in the `Terminating` state using the `{cmdcli}` tool:
```

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc (hunk 2 above)
Comment (jenmalloy) [inline documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]:
```suggestion
is still pending, and you can proceed to the next step.
```

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+In some situations it might be that, due do to infrastructure issues, the clean-up operation
+cannot be performed at this point in time. The `IoTProject` will still be kept, and the
+operator will re-try periodically to clean up the resources. The cleanup will succeed once
+the infrastructure is back in operational state.
+
+In the case that the infrastructure is not expected to function ever again, it might be desirable
+to force the destruction of the `IoTProject` resources.
+
+WARNING: Manually removing the resource cleanup finalizer will skip the cleanup process,
```
[the remark this answers is not in the document]
Comment (ctron) [reply documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]: It depends: If your plan is to fix the infrastructure, and keep it running. Then the database would continue to contain old, unused data, which would not be accessible any more. If you plan on killing the infrastructure, you would continue to have an IoTProject lingering around forver.

File: documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc (hunk 2 above)
Comment (ctron) [reply documentation/modules/proc-iot-troubleshoot-stuck-iotproject.adoc]: Yes, it is the latter.

**Pull request merged**